### PR TITLE
feat(select): use ariakit primitives

### PR DESCRIPTION
**Pull request by bwll-tpio**
- refactor: select component using ariakit primitive
- refactor: updates dir structure stories and types
- test: update tests

## Description of the change

> Description here

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]()

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: packages/components/src/components/select/__tests__/select.test.tsx [hunk 1]
```diff
@@ -24,11 +29,12 @@ describe("Select", () => {
...
-    render(<MockSelect data-testid="select" required />);
-    const renderedSelect = screen.getByTestId("select");
-    expect(renderedSelect.getAttribute("required")).toEqual("");
-  });
+  // eslint-disable-next-line jest/no-commented-out-tests
+  // it("should render a required select", () => {
+  //   render(<MockSelect data-testid="select" required />);
+  //   const renderedSelect = screen.getByTestId("select");
+  //   expect(renderedSelect.getAttribute("required")).toEqual("");
+  // });
```
Comment (richbachman) [inline packages/components/src/components/select/__tests__/select.test.tsx:37]: Are these commented out tests no longer valid?

File: .changeset/large-buckets-camp.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@trueplan/forecast-components": major
+---
+
+[select] updates Select, Option, and OptionGroup to use Select from ariakit. Breaking change with OptionGroup removes the label prop, should now use OptionGroupLabel component as a child
```
Comment (richbachman) [inline .changeset/large-buckets-camp.md:5]: Can we callout the breaking change more? Maybe just add it on a line itself, and make breaking change call caps.

**Comment by richbachman** [comment]
I think the only other thing here is to amend your commit message when you squash and merge this so the breaking change is in the description. See here. https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

**Review by richbachman — APPROVED** [review]
Nice work on this one!

File: .changeset/large-buckets-camp.md (hunk 2 above)
Comment (bwll-tpio) [reply .changeset/large-buckets-camp.md]: fixed

**Pull request merged**